docs/rancher/01-setup-ubuntu/README.md

```diff
@@ -0,0 +1,108 @@
+# How to Set Up and Secure Your Ubuntu Server for Development
+
+## Introduction
+
+This guide will show you how to prepare your Ubuntu server for development by setting up secure access with SSH keys, modifying your SSH configuration for enhanced security, and installing necessary packages through a script.
+
+After completing this guide, you'll be able to:
+
+- Securely log into your server using SSH keys.
+- Configure your SSH settings for improved security.
+- Install essential packages on your server using a script.
+
+## Table of Contents
+
+1. [Introduction](#introduction)
+2. [Prerequisites](#prerequisites)
+3. [Step 1 - Generating and Setting Up SSH Keys](#step-1---generating-and-setting-up-ssh-keys)
+4. [Step 2 - Adjusting Your SSH Settings for Security](#step-2---adjusting-your-ssh-settings-for-security)
+5. [Step 3 - Installing Packages with a Script](#step-3---installing-packages-with-a-script)
+6. [Conclusion](#conclusion)
+
+## Prerequisites
+
+Before starting, you need:
+
+1. An Ubuntu server with root access.
+2. A computer with SSH client software.
+
+## Step 1 - Generating and Setting Up SSH Keys
+
+Create a secure SSH key pair on your computer:
+
+```bash
+ssh-keygen -t rsa -b 4096
+```
+
+Copy your public SSH key to your clipboard for easy access:
+
+```bash
+cat ~/.ssh/id_rsa.pub | pbcopy
+```
+
+Log into your server as the root user:
+
+```bash
+ssh root@<ip_address>
+```
+
+Set up the `.ssh` directory and secure it:
+
+```bash
+mkdir -p ~/.ssh && chmod 700 ~/.ssh
+```
+
+Paste your SSH key into the `authorized_keys` file:
+
+```bash
+vim ~/.ssh/authorized_keys
+```
+
+Then secure the file:
+
+```bash
+chmod 600 ~/.ssh/authorized_keys
+```
+
+**Important:** Test your SSH key by logging in from another terminal tab or session before proceeding to ensure you don't lose access.
```

Comment:
I suggest we place this Important message at the beginning of Step 2, highlighting that the step involves risky behavior, which could result in the loss of SSH connectivity, permanently preventing login.